listenstore/listenstore/listenstore.py

```diff
@@ -119,16 +119,18 @@ def fetch_listens_from_storage(self, user_id, from_id, to_id, limit, order, prec
                      , listen_json
                  WHERE listen.id = listen_json.id
                    AND user_id = :user_id
-                   AND extract(epoch from ts) > :from_id
-                   AND extract(epoch from ts) < :to_id
-              ORDER BY extract(epoch from ts) """ + ORDER_TEXT[order] + """
+                   AND ts > :from_ts
+                   AND ts < :to_ts
+              ORDER BY ts """ + ORDER_TEXT[order] + """
                  LIMIT :limit
             """), {
                 'user_id': user_id,
-                'from_id': from_id,
-                'to_id': to_id,
+                'from_ts': datetime.utcfromtimestamp(from_ts),
+                'to_ts': datetime.utcfromtimestamp(to_ts),
```

Comment:
It is the postgres precision which does this. More specifically, earlier i was using `extract(epoch from ts)` which gave float precision, so typecasting it into int was leading to problems so i used float everywhere. 

But now, since we are switching to using datetime objects into the `listenstore`, then we would have to extract the `epoch` manually everywhere, then we can safely typecast it to int.

This problem can be solved easily once the whole listenstore migration is complete but since the listenstore is not yet complete, hence i can't finish this in this PR. btw, i created a ticket for the same. http://tickets.musicbrainz.org/browse/LB-98